plugin/mutate/source/dextool/plugin/mutate/backend/database/standalone.d

```diff
@@ -279,15 +279,20 @@ struct Database {
     }
 
     /// Returns: the stored scores in ascending order by their `time`.
-    FileScore[] getMutationFileScoreHistory() @trusted {
-        auto app = appender!(FileScore[])();
-        auto stmt = db.prepare(format!"SELECT AVG(score), file_path FROM (SELECT * FROM %s WHERE time_stamp > date('now', '-7 day')) GROUP BY file_path"(
-                mutationFileScoreHistoryTable));
-        foreach (a; stmt.get.execute) {
-            app.put(FileScore(SysTime(0),
-                    typeof(FileScore.score)(a.peek!double(0)), Path(a.peek!string(1))));
+    FileScore[][string] getMutationFileScoreHistory() @trusted {
+        import std.algorithm : sort;
+
+        FileScore[][string] scoreDict;
+        foreach (r; db.run(select!MutationFileScoreHistoryTable)) {
+            scoreDict[r.filePath] ~= FileScore(r.timeStamp,
+                    typeof(FileScore.score)(r.score), Path(r.filePath));
         }
-        return app.data;
+
+        foreach (key; scoreDict.byKey()) {
```

Comment:
what happens if you instead would write a DB query that returned the file score sorted by date?
would the sort here be unnecessary then?